course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,99 @@
+#include "unique_ptr.h"
+#include "unique_ptr.cc"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+template<template<class...> class UniquePtrType>
```

Comment:
I arrived at this via brute force, by trying a bunch of different template definitions (found through google) until my code compiled. I don't quite understand exactly what's going on here, especially the ellipsis (...). If I drop the ellipsis, the code does not compile, but if I also comment out line 92 (unique_ptr_test<std::unique_ptr>();), the code does compile and passes all tests, so it seems like it's only required for std::unique_ptr, and is not necessary for karun::unique_ptr.